File: mobile/src/api/series.ts
import { apiClient as api } from './client';

export interface SeriesSearchResult {
  tmdb_id: number;
  title: string;
  original_title?: string;
  poster_url?: string;
  overview?: string;
  release_date?: string;
  genres: string[];
  vote_average?: number;
}

export interface UserSeriesCreate {
  tmdb_id: number;
  status: 'want_to_watch' | 'watching' | 'watched' | 'dropped';
  rating?: number;
  current_season?: number;
  current_episode?: number;
  last_watched_at?: string;
}

export interface UserSeriesUpdate {
  status?: string;
  rating?: number;
  notes?: string;
  current_season?: number;
  current_episode?: number;
  last_watched_at?: string;
}

export interface UserSeriesResponse {
  id: string;
  status: string;
  rating?: number;
  notes?: string;
  current_season: number;
  current_episode: number;
  last_watched_at?: string;
  tmdb_id: number;
  title: string;
  poster_url?: string;
  release_date?: string;
  total_seasons: number;
  total_episodes: number;
  episode_run_time: number;
  seasons_data: { season_number: number; episode_count: number }[];
}

export const seriesApi = {
  search: async (q: string, year?: number, anime?: boolean): Promise<SeriesSearchResult[]> => {
    const params: any = { q };
    if (year) params.year = year;
    if (anime) params.anime = anime;
    const res = await api.get('/series/search', { params });
    return res.data;
  },
  getMySeries: async (status?: string, limit = 20, offset = 0): Promise<UserSeriesResponse[]> => {
    const res = await api.get('/me/series', { params: { status, limit, offset } });
    return res.data;
  },
  addSeries: async (data: UserSeriesCreate): Promise<UserSeriesResponse> => {
    const res = await api.post('/me/series', data);
    return res.data;
  },
  updateSeries: async (id: string, data: UserSeriesUpdate): Promise<UserSeriesResponse> => {
    const res = await api.patch(`/me/series/${id}`, data);
    return res.data;
  },
  deleteSeries: async (id: string): Promise<void> => {
    await api.delete(`/me/series/${id}`);
  },
};

File: backend/app/schemas/series.py
from pydantic import BaseModel
from datetime import date

class SeriesSearchResult(BaseModel):
    tmdb_id: int
    title: str
    original_title: str | None = None
    poster_url: str | None = None
    overview: str | None = None
    release_date: str | None = None
    genres: list[str] = []
    vote_average: float | None = None

class UserSeriesCreate(BaseModel):
    tmdb_id: int
    status: str = "want_to_watch"
    rating: float | None = None
    current_season: int = 1
    current_episode: int = 0
    last_watched_at: date | None = None

class UserSeriesUpdate(BaseModel):
    status: str | None = None
    rating: float | None = None
    notes: str | None = None
    current_season: int | None = None
    current_episode: int | None = None
    last_watched_at: date | None = None

class UserSeriesResponse(BaseModel):
    id: str
    status: str
    rating: float | None = None
    notes: str | None = None
    current_season: int
    current_episode: int
    last_watched_at: date | None = None
    # Catálogo
    tmdb_id: int
    title: str
    poster_url: str | None = None
    release_date: str | None = None
    total_seasons: int
    total_episodes: int
    episode_run_time: int = 0
    seasons_data: list[dict] = []
